smart-py/marketplace.py

```diff
@@ -53,12 +59,70 @@ def cancel_swap(self, params):
     def update_fee(self, params):
         sp.verify(sp.sender == self.data.manager)
         self.data.fee = params
-        
+
     @sp.entry_point
-    def update_manager(self, params):
+    def update_fee_account(self, params):
         sp.verify(sp.sender == self.data.manager)
+        self.data.fee_account= params
+
+    @sp.entry_point
+    def update_manager(self, params):
+        sp.if sp.now - self.data.last_ping > 100 * 24 * 3600:
+            sp.verify(self.data.secondary_accounts.contains(sp.sender))
+        sp.else:
+            sp.verify(sp.sender == self.data.manager)
+
         self.data.manager = params
-        
+        self.data.secondary_accounts.add(params)
```

Comment:
This leaves the previous manager still part of the set of possible managers. Is there any possible risk or issue with keeping the previous managers in the set?

Reply:
If there is an issue, the new manager can remove him from the set. But to be discussed. We could also not add the current managers to the list. I have no problem about that.

Reply:
what will happen if the old manager comes back after some time?

Reply:
The new manager will have to decide what to do. It could give him back the control, or it could even remove him from the secondary accounts list. The new manager will have total control, but in principle the old manager added him to the secondary_accounts because he could trust on him to be reasonable...

Reply:
Are there also scenarios to consider where the old manager isn't around anymore or cannot be trusted/relied on anymore?

Reply:
The new manager can remove him from the secondary_accounts list and it will be impossible for him to get access manager to the contract, even if the new manager doesn't ping it.
The is always the risk that one of the managers wants to keep total control and removes all secondary accounts. Then in case of bus accident, the contract and the funds will be lost. 
The only option left would be to create another marketplace contract, cancel all swaps, and swap again.
But note that Rafael has better plans for the manager structure (another contract). Have a look at his comment:
https://github.com/hicetnunc2000/smart-contracts/pull/5#issuecomment-874436074